applications/autonomous_worlds_on_substrate.md

```diff
@@ -0,0 +1,346 @@
+
+### Are you intending to raise more funds?
+There are no plans to raise additional funds right now, but we are open to discussing this in the future. The community reception of the game will be a significant factor in determining whether we will seek additional funding. Successful completion of proof-of-concept for the game mechanic will likely lead to further development. Crowd-funding is the most suitable funding model for the future.
+
+## 6. Detailed Development Roadmap :nut_and_bolt:
+
+### Overview
+
+- **Total Estimated Duration:** 12 weeks
+- **Full-Time Equivalent (FTE):**  3
+- **Total Costs:** 9900 DOT ~ $75,000  USD
+
+### Milestone 1: Light Client + Relayer
+
+- **Estimated duration:** 4 weeks
+- **FTE:**  3
+- **Costs:** 3300 DOT (6.875 DOT/hr)
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Specify the IP owning entity; ensure proper copyright compliance for all reused materials, including appropriate licenses and attributions, or choose an open-source license (Apache 2.0, GPLv3, MIT, Unlicense). |
+| **0b.** | Documentation | Provide comprehensive **inline documentation** of the code and a detailed **tutorial**. The tutorial should guide users on how to set up, play the game, and assess the milestone's deliverables, ensuring functionality and compliance with the milestone objectives. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Platform | Provide a detailed description of the game's platform compatibility and the infrastructure setup required to host and run the game, including supported operating systems, hardware requirements, and necessary backend services. |
+| 0e. | Article | We will publish an **article**/workshop that explains [...] (what was done/achieved as part of the games bounty). (Content, language, and medium should reflect your target audience described above.) |
+| 1. | Ideal Network - Runtime and Pallets Upgrade (2 weeks) | We modify the Ideal Network runtime to enable the "Smart Contract Light Client". This includes creating a new pallet to store cryptographic commitments required for efficient onchain verification of ETF justifications with the smart contract.  |
+| 2. | Centralized Relayer Component (1 week) | We implement a centralized relayer which is responsible for subscribing to ETF justifications, interpolating signatures and aggregating proofs, and sending the aggregated messages to the SCLC. |
+| 3. | Smart Contract: The "beacon" Contract (1 week) | We develop and deploy the actual smart contract light client. |
+| 4. | Demo Contract: Rock Paper Scissors **Part 1** (1 week) | We develop a rock-paper-scissors demo contract that consumes randomness from the SCLC and uses it in a meaningful way (to play p2p non-interactive RPS). |

```

Comment:
Do we need a Demo? If we do, does it make sense to make something simpler than Rock Paper Scissors?

Reply:
Maybe leave it as a generic "Demo" as in milestone 2?

Reply:
I like the idea of each milestone having a demo as an output, something tangible and usable. I'm actually not 100% comfortable with how generic milestone 2's final outcome is. We could even go as far as hosting the site (I already have one started here: http://rps.idealabs.network). RPS is actually really easy to implement, we pretty much just need to get the signature bytes, sum them together, then take that mod 3 and compare to user input.

Reply:
Fair enough, sounds good to me